pkg/ipblock/ipblock.go

```diff
@@ -209,42 +210,31 @@ func NewIPBlockFromCidrOrAddress(s string) (*IPBlock, error) {
 
 // NewIPBlockFromCidrList returns IPBlock object from multiple CIDRs given as list of strings
 func NewIPBlockFromCidrList(cidrsList []string) (*IPBlock, error) {
-	res := &IPBlock{ipRange: interval.CanonicalSet{}}
+	ipRange := interval.NewCanonicalIntervalSet()
 	for _, cidr := range cidrsList {
 		block, err := NewIPBlockFromCidr(cidr)
 		if err != nil {
 			return nil, err
 		}
-		res = res.Union(block)
+		ipRange = ipRange.Union(block.ipRange)
 	}
-	return res, nil
+	return &IPBlock{ipRange: *ipRange}, nil
 }
 
-// NewIPBlock returns an IPBlock object from input cidr str an exceptions cidr str
-func NewIPBlock(cidr string, exceptions []string) (*IPBlock, error) {
-	res := IPBlock{ipRange: interval.CanonicalSet{}}
-	span, err := cidrToInterval(cidr)
-	if err != nil {
-		return nil, err
```

Comment:
I would keep this function (`func NewIPBlock(cidr string, exceptions []string)`)
It is used by another repo, also planned to import this pkg now

Reply:
I could not find the usage

Reply:
We can add `ExceptCidrs()` function instead:
```go
// ExceptCidrs returns a new IPBlock with all cidr ranges removed 
func (b *IPBlock) ExceptCidrs(cidrExceptions ...string) (*IPBlock, error) {
	res := b.Copy()
	for i := range cidrExceptions {
		hole, err := NewIPBlockFromCidr(cidrExceptions[i])
		if err != nil {
			return nil, err
		}
		res = res.Subtract(hole)
	}
	return res, nil
}
```

Reply:
> I could not find the usage

It's in another repo (not vpc analysis/snyhtesis) that will also import models , for example:
https://github.com/np-guard/netpol-analyzer/blob/main/pkg/netpol/eval/internal/k8s/netpol.go#L318

Reply:
Ok, good to know... I will add it to my workspace and see.